pkg/logql/downstream.go

```diff
@@ -466,6 +504,61 @@ func ConcatEvaluator(evaluators []StepEvaluator) (StepEvaluator, error) {
 	)
 }
 
+func TopkMergeEvalator(evaluators []ProbabilisticStepEvaluator) (StepEvaluator, error) {
+	return NewStepEvaluator(
+		func() (ok bool, ts int64, vec promql.Vector) {
+			var cur StepResult
+			var acc *sketch.Topk
+			for _, eval := range evaluators {
+				ok, ts, cur = eval.Next()
+				if acc == nil {
+					acc = cur.TopkVector().Topk
+				} else {
+					acc.Merge(cur.TopkVector().Topk)
+				}
+			}
+
+			for _, r := range acc.Topk() {
+				for _, e := range r.Result {
+					// TODO(karsten): handle error.
+					ls, _ := syntax.ParseLabels(e.Event)
```

Comment:
at this point the labels would be labels that have already been stored in loki or parsed from log lines, so they should be okay? we could treat an error here as non-fatal imo, be more strict about what's acceptable to pass to `topk.Observe` in the first place?

Reply:
It's tricky. `topk.Observe` could use the label hash to begin with and we could store the labels differently so there's no need for parsing. However, that's quite some work. We have to handle the error somehow so that we catch bugs.